app/containers/UIKit/Image.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+import FastImage from 'react-native-fast-image';
+import PropTypes from 'prop-types';
+import { BLOCK_CONTEXT } from '@rocket.chat/ui-kit';
+
+import { MessageImage } from '../message/Image';
+
+const styles = StyleSheet.create({
+	image: {
+		borderRadius: 2
+	},
+	mediaContext: {
+		marginRight: 8
+	}
+});
+
+const ThumbContext = args => <View style={styles.mediaContext}><Thumb size={20} {...args} /></View>;
+
+export const Thumb = ({ element, size = 88 }) => (
+	<FastImage
+		style={[{ width: size, height: size }, styles.image]}
+		source={{ uri: element.imageUrl }}
+	/>
+);
+Thumb.propTypes = {
+	element: PropTypes.object,
+	size: PropTypes.number
+};
+
+export const Media = ({ element, theme }) => (
+	<MessageImage
+		img={element.imageUrl}
+		theme={theme}
+	/>
+);
+Media.propTypes = {
+	element: PropTypes.object,
+	theme: PropTypes.string
+};
+
+const genericImage = (element, context, theme) => {
```

Comment:
Should this open our image viewer?

Reply:
I think yes!
Done.
![](https://media.giphy.com/media/TH6d1kAfYfx7V3nfoc/giphy.gif)